### PR TITLE
Clean up GL backend code

### PR DESCRIPTION
**Pull request by tronical**
Remove GLRenderer and GLRendererData. Most of the fields fold nicely
into the MappedWindow.

### CODE REVIEW

File: sixtyfps_runtime/corelib/item_rendering.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ impl CachedRenderingData {
     pub fn get_or_update<T: Clone>(
         &self,
         cache: &RefCell<RenderingCache<T>>,
-        update_fn: impl FnOnce() -> T,
+        update_fn: impl FnMut() -> T,
```
Comment (ogoffart) [inline sixtyfps_runtime/corelib/item_rendering.rs:41]: Why the change?  Isn't  FnOnce stronger than FnMut ?

File: sixtyfps_runtime/rendering_backends/gl/lib.rs [hunk 2]
```diff
@@ -876,13 +780,13 @@ impl ItemRenderer for GLItemRenderer {
                     let shadow_image_height = shadow_rect.max_y().ceil() as usize;
 
                     let shadow_image = CachedImage::new_empty_on_gpu(
-                        &self.shared_data.canvas,
+                        &self.canvas,
```
Comment (ogoffart) [inline sixtyfps_runtime/rendering_backends/gl/lib.rs:783]: There is also a local `canvas` variable. Should that one not be used? Or what's the difference

**Review by ogoffart — APPROVED** [review]
after fixing the warning

File: sixtyfps_runtime/rendering_backends/gl/lib.rs (hunk 2 above)
Comment (tronical) [reply sixtyfps_runtime/rendering_backends/gl/lib.rs:783]: Well spotted, yes, that should be used instead.

File: sixtyfps_runtime/rendering_backends/gl/lib.rs [hunk 3]
```diff
@@ -1029,20 +926,17 @@ impl ItemRenderer for GLItemRenderer {
         {
             let paint = layer_to_restore.image.as_paint();
 
-            self.shared_data
-                .layer_images_to_delete_after_flush
-                .borrow_mut()
-                .push(layer_to_restore.image);
+            self.layer_images_to_delete_after_flush.push(layer_to_restore.image);
```
Comment (tronical) [inline sixtyfps_runtime/rendering_backends/gl/lib.rs:929]: Removing this use of interior mutability is what triggers the change from `FnOnce` to `FnMut`. If you prefer then I can keep that vector in a RefCell instead and stay with `FnOnce`.

File: sixtyfps_runtime/corelib/item_rendering.rs (hunk 1 above)
Comment (tronical) [reply sixtyfps_runtime/corelib/item_rendering.rs:41]: See comment in `lib.rs`, there this allows replacing a use of interior mutability. But maybe that's not worth it and it's better to keep this `FnOnce`?

Comment (tronical) [reply sixtyfps_runtime/corelib/item_rendering.rs:41]: I think I spoke too soon. Not sure why, but if I change it back now it works :)

File: sixtyfps_runtime/rendering_backends/gl/lib.rs (hunk 3 above)
Comment (tronical) [reply sixtyfps_runtime/rendering_backends/gl/lib.rs:929]: In the forest of warnings I overlooked the one that really caused it, which was an immutable borrow of self outside the closure, that didn't go well with the mixture of immutable and mutable borrows inside. But fortunately that was easy to solve :)

**Review by ogoffart — APPROVED** [review]
